pkg/bff/organization.go

```diff
@@ -47,3 +49,45 @@ func (s *Server) OrganizationFromClaims(c *gin.Context) (org *models.Organizatio
 
 	return org, nil
 }
+
+// AddOrganizationCollaborator adds a new collaborator record to an organization.
+func AddOrganizationCollaborator(org *models.Organization, email string) (err error) {
+	if email == "" {
+		return errors.New("email address is required to add an organization collaborator")
+	}
+
+	if org.Collaborators == nil {
+		org.Collaborators = make(map[string]*models.Collaborator)
+	}
+
+	// Don't overwrite an existing collaborator
+	if _, ok := org.Collaborators[email]; ok {
+		return fmt.Errorf("collaborator with email address %s already exists", email)
```

Comment:
This will return a bad request to the front end in the current workflow; please make sure that Cletus understands that he will have to handle this error as a warning on the front-end.